views/components/ColorPicker.jsx

```diff
@@ -1,39 +1,55 @@
-import { Paper, Typography } from "@material-ui/core"
-import React, { useCallback, useRef, useState } from "react"
-import { HexColorPicker } from "react-colorful"
+import { Paper, Typography } from '@material-ui/core'
+import React, { useCallback, useRef, useState } from 'react'
+import { HexColorPicker } from 'react-colorful'
 
-import useClickOutside from "../hooks/useClickOutside"
+import useClickOutside from '../hooks/useClickOutside'
 
-const ColorPicker = ({ 
-  classes, 
-  color, 
-  onColorChange, 
-  idx
+const ColorPicker = ({
+  classes,
+  color,
+  handleColorChange,
+  idx,
+  presetColors,
 }) => {
-  const colorKey = 'color'
   const [isColorPickerOpen, setColorPickerToggle] = useState(false)
   const popover = useRef()
   const close = useCallback(() => setColorPickerToggle(false), [])
   useClickOutside(popover, close)
 
   return (
     <div className={classes.swatchContainer}>
-      <Typography variant='caption' className={classes.colorLabel}>
-          Color
+      <Typography variant="caption" className={classes.colorLabel}>
+        Color
       </Typography>
       <Paper
         className={classes.swatch}
         style={{ backgroundColor: color }}
         onClick={() => setColorPickerToggle(true)}
       />
-    {isColorPickerOpen && (
-      <div className={classes.popover} ref={popover}>
-        <HexColorPicker 
-          color={color} 
-          onChange={(newColor) => onColorChange({target: { name: colorKey, value: newColor }}, idx)} 
-        />
-      </div>
-    )}
+      {isColorPickerOpen && (
+        <div className={classes.popover} ref={popover}>
+          <HexColorPicker
+            color={color}
+            onChange={(newColor) => handleColorChange(newColor, idx)}
+          />
+          <div className={classes.colorPickerSwatchBox}>
+            {presetColors.map((presetColor) => (
+              <input
```

Comment:
this input needs `key={presetColor}`

Reply:
https://github.com/AMP-SCZ/dpdash/pull/183/commits/5f34f25c9f55743482b2461da6f5a897a9a92a81